src/redis-check-rdb.c

```diff
@@ -272,6 +273,21 @@ int redis_check_rdb(char *rdbfilename, FILE *fp) {
             decrRefCount(auxkey);
             decrRefCount(auxval);
             continue; /* Read type again. */
+        } else if (type == RDB_OPCODE_MODULE_AUX) {
+            /* AUX: Auxiliary data for modules. */
+            uint64_t moduleid, when_opcode, when;
+            rdbstate.doing = RDB_CHECK_DOING_READ_MODULE_AUX;
```

Comment:
@oranagra do you refer to this?

Reply:
yes, that "enum" is used to look up a string from a table later on.
so adding an index without extending the array would cause an invalid read from the array.

Reply:
Got it